commit 207ab5d5a250a738b261529e05ea65815c6262c4
Author: Florian Fainelli <f.fainelli@gmail.com>
Date:   Tue Aug 22 15:24:47 2017 -0700

    fsl/man: Inherit parent device and of_node
    
    
    [ Upstream commit a1a50c8e4c241a505b7270e1a3c6e50d94e794b1 ]
    
    Junote Cai reported that he was not able to get a DSA setup involving the
    Freescale DPAA/FMAN driver to work and narrowed it down to
    of_find_net_device_by_node(). This function requires the network device's
    device reference to be correctly set which is the case here, though we have
    lost any device_node association there.
    
    The problem is that dpaa_eth_add_device() allocates a "dpaa-ethernet" platform
    device, and later on dpaa_eth_probe() is called but SET_NETDEV_DEV() won't be
    propagating &pdev->dev.of_node properly. Fix this by inherenting both the parent
    device and the of_node when dpaa_eth_add_device() creates the platform device.
    
    Fixes: 3933961682a3 ("fsl/fman: Add FMan MAC driver")
    Signed-off-by: Florian Fainelli <f.fainelli@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/freescale/fman/mac.c b/drivers/net/ethernet/freescale/fman/mac.c
index 6e67d22fd0d5..1c7da16ad0ff 100644
--- a/drivers/net/ethernet/freescale/fman/mac.c
+++ b/drivers/net/ethernet/freescale/fman/mac.c
@@ -623,6 +623,8 @@ static struct platform_device *dpaa_eth_add_device(int fman_id,
 		goto no_mem;
 	}
 
+	pdev->dev.of_node = node;
+	pdev->dev.parent = priv->dev;
 	set_dma_ops(&pdev->dev, get_dma_ops(priv->dev));
 
 	ret = platform_device_add_data(pdev, &data, sizeof(data));
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1006_207ab5d5a250a738b261529e05ea65815c6262c4.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1006_207ab5d5a250a738b261529e05ea65815c6262c4.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
